### PR TITLE
Support toUInt8/toInt8 for if constant condition optimization.

### LINKED ISSUE #31687: Missing unused column in "if" expression when upgrading to 21.11.4.14
21.11.4.14
```
create table test (x String) Engine=StripeLog
select if(toUInt8(0), y, 42) from test
```

Throws exception `DB::Exception: Missing columns: 'y'`
Excpecting 42
btw `select if(0, y, 42) from test` works fine

### PR DESCRIPTION
**Pull request by KochetovNicolai**
Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Fixes #31687

### CODE REVIEW

File: src/Interpreters/OptimizeIfWithConstantConditionVisitor.cpp [hunk 1]
```diff
@@ -49,6 +49,16 @@ static bool tryExtractConstValueFromCondition(const ASTPtr & condition, bool & v
                 }
             }
         }
+        else if (function->name == "toUInt8" || function->name == "toInt8")
```
Comment (alexey-milovidov) [inline src/Interpreters/OptimizeIfWithConstantConditionVisitor.cpp:52]: Do we still need `OptimizeIfWithConstantCondition`?
Also maybe make a setting to disable this optimization?

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]
The change looks ad-hoc.

**Comment by alexey-milovidov** [comment]
What if any other function will be used?

File: src/Interpreters/OptimizeIfWithConstantConditionVisitor.cpp (hunk 1 above)
Comment (KochetovNicolai) [reply src/Interpreters/OptimizeIfWithConstantConditionVisitor.cpp:52]: Looks like we need it, there is no other way to skip unused columns

Comment (alexey-milovidov) [reply src/Interpreters/OptimizeIfWithConstantConditionVisitor.cpp:52]: Where exactly it is needed?

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**